daemon/containerd/image_push.go

```diff
@@ -183,6 +215,92 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 	return nil
 }
 
+func (i *ImageService) getPushDescriptor(ctx context.Context, img containerdimages.Image, platform *ocispec.Platform) (ocispec.Descriptor, error) {
+	hostPlatform := i.hostPlatform()
+	pm := matchAllWithPreference(hostPlatform)
+	if platform != nil {
+		pm = platforms.OnlyStrict(*platform)
+	}
+
+	anyMissing := false
+
+	var bestMatchPlatform ocispec.Platform
+	var bestMatch *ImageManifest
+	var presentMatchingManifests []*ImageManifest
+	err := i.walkReachableImageManifests(ctx, img, func(im *ImageManifest) error {
+		available, err := im.CheckContentAvailable(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to determine availability of image manifest %s: %w", im.Target().Digest, err)
+		}
+
+		if !available {
+			anyMissing = true
+			return nil
+		}
+
+		if im.IsAttestation() {
+			return nil
+		}
+
+		imgPlatform, err := im.ImagePlatform(ctx)
+		if err != nil {
```

Comment:
Do we need to handle context cancel/timeout errors here?

Reply:
We don't need to I think. If context is cancelled then `ImagePlatform` will return with an error, which will then cause the `walkImageManifests` to also return early with an error.

Reply:
We rarely do that in the rest of the codebase. Also, the time window here is pretty small so it's unlikely that the cancellation will actually happen in this code for the most of the time.

In general there's a lot of calls that can error out with a context cancellation and handling it in every call to containerd wouldn't be too practical. Perhaps it would be better to handle (prefix the error with "operation cancelled at:"?) somewhere higher up the stack (like the API)?